docs/core/diagnostics/dotnet-dsrouter.md

```diff
@@ -0,0 +1,262 @@
+
+### Synopsis
+
+```console
+dotnet-dsrouter client-client
+    [-ipcc|--ipc-client <ipcClient>]
+    [-tcpc|--tcp-client <tcpClient>]
+    [-rt|--runtime-timeout <timeout>]
+    [-v|--verbose <level>]
+    [-fp|--forward-port <platform>]
+```
+
+### Options
+
+- **`-ipcc, --ipc-client <ipcClient>`**
+  The diagnostic tool diagnostics server ipc address (--diagnostic-port argument). Router connects diagnostic tool ipc server when establishing a new route between runtime and diagnostic tool.
+
+- **`-tcpc, --tcp-client <tcpClient>`**
+  The runtime TCP/IP address using format [host]:[port]. Router can connect 127.0.0.1, [::1], ipv4 address, ipv6 address, hostname addresses. Launch runtime using `DOTNET_DiagnosticPorts` environment variable to setup listener
+
+- **`-rt, --runtime-timeout <runtimeTimeout>`**
+  Automatically shut down router if no runtime connects to it before specified timeout (seconds). If not specified, router won't trigger an automatic shutdown.
+
+- **`-v, --verbose <verbose>`**
+  Enable verbose logging (debug|trace)
+
+- **`-fp, --forward-port <forwardPort>`**
+  Enable port forwarding, values Android|iOS for TcpClient and only Android for TcpServer. Make sure to set `ANDROID_SDK_ROOT` before using this option on Android.
+
+## Collect a startup trace using dotnet-trace from a .NET application running on Android
+
+Sometimes it may be useful to collect a trace of an application from its startup. Below illustrates the process of doing so targeting a .NET application running on Android. Since `dotnet-dsrouter` is run using port forwarding, the same scenario will work against applications running in local emulator as well as physical device attached over USB. Make sure to set `ANDROID_SDK_ROOT` before using this option or `dotnet-dsrouter` won't be able to find `adb` needed to setup port forwarding.
+
+- Launch dotnet-dsrouter in server-server mode:
+
+  ```console
+  dotnet-dsrouter server-server -ipcs ~/mylocalport -tcps 127.0.0.1:9000 --forward-port Android &
+  ```
+
+- Set `DOTNET_DiagnosticPorts` environment variable using `AndroidEnvironment`:
+
+  Create a file in same folder as `.csproj` using a name like `app.env`, add environment variable into file, `DOTNET_DiagnosticPorts=127.0.0.1:9000,suspend` and include following `ItemGroup` into `.csproj`:
+
+  ```xml
+  <ItemGroup Condition="'$(AndroidEnableProfiler)'=='true'">
+    <AndroidEnvironment Include="app.env" />
+  </ItemGroup>
+  ```
+
+  It is also possible to set DOTNET_DiagnosticPorts using `adb shell setprop`:
+
+  ```console
+  adb shell setprop debug.mono.profile '127.0.0.1:9000,suspend'
+  ```
+
+- Build and launch application using .NET Android SDK and enable tracing by passing `/p:AndroidEnableProfiler=true` to MSBuild. Since application have been configured to suspend on startup, it will connect back to `dotnet-dsrouter` TCP/IP listener running on 127.0.0.1:9000 and wait for diagnostic tooling to connect, before resuming application execution.
```

Comment:
```suggestion
- Build and launch application using .NET Android SDK and enable tracing by passing `/p:AndroidEnableProfiler=true` to MSBuild. Since application have been configured to suspend on startup, it will connect back to `dotnet-dsrouter` TCP/IP listener running on `127.0.0.1:9000` and wait for diagnostic tooling to connect, before resuming application execution.
```